app/(company)/declare/components/SignForm.js

```diff
@@ -0,0 +1,173 @@
+'use client';
+
+import { useState } from 'react';
+import { isValidEmail } from '@shared/inputs/EmailInput';
+import { Tangerine } from '@next/font/google';
+import Link from 'next/link';
+import BaseButtonClient from '@shared/buttons/BaseButtonClient';
+import gpApi from 'gpApi';
+import gpFetch from 'gpApi/gpFetch';
+import Image from 'next/image';
+
+const tangerine = Tangerine({
+  weight: ['400', '700'],
+  subsets: ['latin'],
+});
+
+export async function subscribeEmail(payload) {
+  try {
+    await gpFetch(gpApi.homepage.subscribeEmail, payload);
+    return true;
+  } catch (e) {
+    console.log('error', e);
+    return false;
+  }
+}
+
+export default function SignatureForm({
+  formId,
+  pageName,
+  label = 'SIGN NOW',
+  labelId,
+}) {
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
```

Comment:
this is fine, but you can set all of these in one state as I do in all other forms